Draft css styleguide

msecret opened the pull request:
Draft 0.0.1 of css guidelines

Comment on doc here: https://docs.google.com/a/gsa.gov/document/d/1t6Rjslq9I20reFF3xfW3FVeV2NWFaDKauDj_LQJC5mc/edit?usp=sharing

```
TODO:
```
- Another architecture type.
- An example of everything.
- More rationale.
- Ensure consistency.

Commit:
Start version of css styleguide

WIP

Commit:
Draft 0.0.1 of css guidelines

TODO:
- Another architecture type.
- An example of everything.
- More rationale.
- Ensure consistency.

msecret commented on css-styleguide.md:
```diff
@@ -0,0 +1,652 @@
+# Coding styleguide
+
+1. Preproocessor
```
Make these actual inks in Markdown

msecret commented on css-styleguide.md:
```diff
@@ -0,0 +1,652 @@
+  margin: 3px;
+}
+
+// Bad
```
Consistency: move // Bad examples before // Good

msecret commented on css-styleguide.md:
````diff
@@ -0,0 +1,652 @@
+explicitly set all the available values.
+```scss
+// Bad
+margin: inherit 3em;
````
Line break after this, before // Good

msecret commented on css-styleguide.md:
```diff
@@ -0,0 +1,652 @@
+  width: (100% / 3);
+}
+
+// Bad
```
Bad example before good.

msecret commented on css-styleguide.md:
```diff
@@ -0,0 +1,652 @@
+	2. @extend directives
+	3. @include directives
+	4. properties
+- For properties alphabetical order or type order, just keep the whole project consistant.
```
Possibly need to explain type vs alphabetical order more in depth with examples.

msecret commented on css-styleguide.md:
```diff
@@ -0,0 +1,652 @@
+
+## Units
+### Measurements
+- Use **rem** units for font sizes with a em fallback.
```
I've done this on Peace Corps and it's worked well so far, but I'd like more research done before it's implemented as a guide.

msecret commented on css-styleguide.md:
````diff
@@ -0,0 +1,652 @@
+```
+- Classes should be lowercase.
+- Avoid camelcase.
+- Ensure things are understanbly named.
````
"understandably"

msecret commented on css-styleguide.md:
````diff
@@ -0,0 +1,652 @@
+  }
+}
+```
+- Use mixins for compoennts to change size.
````
typo "components"

msecret commented on css-styleguide.md:
```diff
@@ -0,0 +1,652 @@
+- Move next to more specific classes, often with nesting.
+- Move next to overrides, possibly with !important rules.
+- Import alphabetically
+- Only modify import order for groups of files, not specific files.
```
Having trouble explaining this.

msecret commented on css-styleguide.md:
````diff
@@ -0,0 +1,652 @@
+- Place media queries nearest to the class they are affecting.
+- Rather the focusing on devices when deciding where to put breakpoints, focus
+on content.
+```scss
````
Needs "bad" and "good" label.

meiqimichelle commented on css-styleguide.md:
```diff
@@ -0,0 +1,652 @@
+- Full example
+
+## Preprocessor
+The  most supported preprocessor is sass/scss. Using this pre-processor means you'll get supported resources such as frameworks, libraries, tutorials and a comprehensive styleguide as support.
```
Do you mean the most supported preprocessor at 18F or in general? Agree that this the most supported one at 18F, if that's what you mean, and that this is a good way to suggest that people consider using it ("this means you'll get XYZ").

Commit:
Added table of contents with links

Checked that all links work.

Commit:
Incorporated @mhz comments

Addressed and Incorporated all comments from Michelle that were put in the
Google Doc by Feb 27 2015.

Added new sections for notes under Format, added section on how to setup
various text editors to use 80 chars per line.

Commit:
Continued edits

Various edits pulled from the gdoc.

Commit:
Fixed formatting of code blocks

To be compatible with git-flavored markdown, the code blocks each need a new
line before and after to ensure the correct formatting.

This code change was not put into the google doc yet. This still needs to be done.

msecret commented:
Moving here for better organization: https://github.com/18F/frontend/pull/15
